src/ocean_emulators/utils/data.py

```diff
@@ -115,6 +132,16 @@ def normalize_with(
         norm = norm.to(data.dtype)
         return norm
 
+    @classmethod
+    def from_dataset(cls, dataset: xr.Dataset) -> Self:
```

Comment:
🐑  optional: let users specify a name of the dataset, e.g. 
```suggestion
    def from_dataset(cls, dataset: xr.Dataset, *, name: str = 'dataset') -> Self:
```

...to be used as the data class name below. I could see how this would be helpful when reading the data class name in the debugger.

Reply:
Done, thanks